safe-hts-precompile/SafeHTS.sol

```diff
@@ -0,0 +1,458 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.5.0 <0.9.0;
+pragma experimental ABIEncoderV2;
+
+import "../hts-precompile/IHederaTokenService.sol";
+import "./HederaResponseCodes.sol";
+
+library SafeHTS {
+
+    address constant precompileAddress = address(0x167);
+    // 90 days in seconds
+    uint32 constant defaultAutoRenewPeriod = 7776000;
+
+    function safeCryptoTransfer(IHederaTokenService.TokenTransferList[] memory tokenTransfers) internal {
+        int32 responseCode;
+        (bool success, bytes memory result) = precompileAddress.call(
+            abi.encodeWithSelector(IHederaTokenService.cryptoTransfer.selector, tokenTransfers));
+        responseCode = success ? abi.decode(result, (int32)) : HederaResponseCodes.UNKNOWN;
+        require(responseCode == HederaResponseCodes.SUCCESS, "Safe crypto transfer failed!");
+    }
+
+    function safeMintToken(IHederaTokenService token, uint64 amount, bytes[] memory metadata) internal
```

Comment:
These should all go back to address. IHEderaTokenService is the interface for the precompile, not the type of the tokens.  These contracts are untyped.

Reply:
Done.

Reply:
If the goal is to mimic OpenZeppelin's SafeERC20 then the name should stutter.  Example - https://github.com/code-423n4/2022-05-alchemix/blob/71abbe683dfd5c0686b7e594fb4f78a14b668d8b/contracts-full/adapters/fuse/FuseTokenAdapterV1.sol#L70